solutions/ex2/Main.hs

```diff
@@ -389,4 +391,66 @@ instance MakeClearVariableStatement MJavaSig where
 
 
 exampleLuaProgram :: IO (Maybe (MLuaTerm LBlockL))
-exampleLuaProgram = parseFile @MLuaSig "input-files/lua/Foo.lua"
\ No newline at end of file
+exampleLuaProgram = parseFile @MLuaSig "input-files/lua/Foo.lua"
+
+-- Hints about supporting Lua:
+-- Create a new typeclass that says that `fs` can make Block from BlockItems somehow, not necessarily via SimpleBlock,
+-- and allowing the user to specify a non-standard "block end" term.
+-- Use UndecidableInstances as there are too many constraints.
+
+class (Block :-<: fs, All HFunctor fs) => CanMakeBlockFromItems fs where
+  makeBlockFromItems :: Term fs [BlockItemL] -> Term fs BlockEndL -> Term fs BlockL
+
+type CanClearVariablesGen fs =
+  ( All HTraversable fs,
+    ExtractF [] (Term fs),
```

Comment:
Group HTraversable, HFoldable, and HFunctor together. Group ExtractF and InsertF together.

Reply:
Group the fragment inclusions (:-<:) together as well.

Reply:
Note the formatting of CanClearVariables earlier in this file -- good practice to keep. Or at least one that I'd like to request be done consistently in this tutorial.